### PR TITLE
feat(git): allow custom merge commit ids

### LINKED ISSUE #3356: Allow tags and IDs on merge commits in gitgraph diagrams
As of now, it seems that it is not possible to give a merge commit an ID and/or a tag. But git allows is and it sometime is important to do this in a gitgraph diagram to explain or demonstrate stuff.

### PR DESCRIPTION
**Pull request by aloisklink**
## :bookmark_tabs: Summary

Currently, merge commits can have a git tag, but they cannot have a custom git commit ID.

This commit allows modifying the default merge commit id.

Resolves #3356

Partially resolves issue #3238 (most of the discussion there is talking about merge commit IDs, but technically the issue is asking for all merge commits to have all the same options as base commits).

## :straight_ruler: Design Decisions

In order to view the custom merge commit ids, I've undone commit https://github.com/mermaid-js/mermaid/commit/3ccf027f4285ba89bba13de62d05d0ce58ac79b5.

**This means that all merge commits have their IDs displayed, including auto-generated commit IDs.** Personally, I think this is okay, since it's how git works, but it does add a bit of clutter to the git graphs.

If you want, I could hide merge commits labels behind a config option `gitGraphConfig.showMergeCommitLabel`.

I've also made a commit (https://github.com/aloisklink/mermaid/commit/954bf9e1607afc0bcff5166199d5a85b537e5acc), where I update some ```mermaid-examples``` diagrams in the docs, but they're not included in this PR, since the docs currently only use v9.1.5 of mermaid: https://github.com/mermaid-js/mermaid/blob/cde3a7cf70cbf6f33406ebb4e83e58d7af9feb7d/docs/index.html#L21-L22

### Example changes:

```mermaid-example
%%{init: { 'logLevel': 'debug', 'theme': 'forest' } }%%
      gitGraph
        commit
        branch hotfix
        checkout hotfix
        commit
        branch develop
        checkout develop
        commit id:"ash" tag:"abc"
        branch featureB
        checkout featureB
        commit type:HIGHLIGHT
        checkout main
        checkout hotfix
        commit type:NORMAL
        checkout develop
        commit type:REVERSE
        checkout featureB
        commit
        checkout main
        merge hotfix
        checkout featureB
        commit
        checkout develop
        branch featureA
        commit
        checkout develop
        %% ID added here
        merge hotfix id: "bca"
        checkout featureA
        commit
        checkout featureB
        commit
        checkout develop
        %% Tag added here (already working before)
        merge featureA tag: "v1.1.0-rc1"
        branch release
        checkout release
        commit
        checkout main
        commit
        checkout release
        merge main
        checkout develop
        merge release
```

image

### :clipboard: Tasks
Make sure you
- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md) 
- [x] :computer: have added unit/e2e tests (if appropriate) 
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

File: src/diagrams/git/gitGraphParserV2.spec.js [hunk 1]
```diff
@@ -496,6 +496,76 @@ describe('when parsing a gitGraph', function () {
...
+      commit
+      checkout main
+      %% Merge ID and Tag (reverse order)
+      merge testBranch2 id: "4-444" tag: "merge-tag2"
+      branch testBranch3
+      checkout testBranch3
+      commit
+      checkout main
+      %% just Merge ID
+      merge testBranch3 id: "6-666"
```
Comment (sidharthv96) [inline src/diagrams/git/gitGraphParserV2.spec.js:519]: Is just Tag a valid scenario?

Comment (aloisklink) [reply src/diagrams/git/gitGraphParserV2.spec.js:519]: Yep, it's in a previous test (added in https://github.com/mermaid-js/mermaid/pull/2955): https://github.com/mermaid-js/mermaid/blob/cde3a7cf70cbf6f33406ebb4e83e58d7af9feb7d/src/diagrams/git/gitGraphParserV2.spec.js#L464-L472 

Maybe it's worth potentially merging these tests together, instead of having a separate one just for merge ids.

Comment (sidharthv96) [reply src/diagrams/git/gitGraphParserV2.spec.js:519]: Cool!
More tests are fine. We don't know what edge case we might miss.

**Review by sidharthv96 — APPROVED** [review]

**Comment by aloisklink** [comment]
> We had already started working on root issue https://github.com/mermaid-js/mermaid/issues/3238 to enhance the merge commit to allow all commit-specific attributes, including "id," "tag" and "type".

That's my fault for not looking at #3238 and seeing that you were assigned to it (I only spotted that issue after I had made my PR).
Still, I'm happy you at least managed to reuse some of my test cases/documentation!
